manifests/init.pp

```diff
@@ -461,94 +461,93 @@
-  Boolean $use_systemd                                                  = $apache::params::use_systemd,
-  Hash $mime_types_additional                                           = $apache::params::mime_types_additional,
-  String $file_mode                                                     = $apache::params::file_mode,
-  Array $root_directory_options                                         = $apache::params::root_directory_options,
-  Boolean $root_directory_secured                                       = false,
-  String $error_log                                                     = $apache::params::error_log,
-  String $scriptalias                                                   = $apache::params::scriptalias,
-  String $access_log_file                                               = $apache::params::access_log_file,
-  Array[Enum['h2', 'h2c', 'http/1.1']] $protocols                = [],
-  Optional[Boolean] $protocols_honor_order                       = undef,
+  String $apache_name                                                        = $apache::params::apache_name,
+  String $service_name                                                       = $apache::params::service_name,
+  Variant[Array, Boolean] $default_mods                                      = true,
+  Boolean $default_vhost                                                     = true,
+  Optional[String] $default_charset                                          = undef,
+  Boolean $default_confd_files                                               = true,
+  Boolean $default_ssl_vhost                                                 = false,
+  Stdlib::Absolutepath $default_ssl_cert                                     = $apache::params::default_ssl_cert,
+  Stdlib::Absolutepath $default_ssl_key                                      = $apache::params::default_ssl_key,
+  Optional[Stdlib::Absolutepath] $default_ssl_chain                          = undef,
+  Optional[Stdlib::Absolutepath] $default_ssl_ca                             = undef,
+  Optional[Stdlib::Absolutepath] $default_ssl_crl_path                       = undef,
+  Optional[Stdlib::Absolutepath] $default_ssl_crl                            = undef,
+  Optional[String] $default_ssl_crl_check                                    = undef,
+  Boolean $default_ssl_reload_on_change                                      = false,
+  String $default_type                                                       = 'none',
+  Optional[Variant[Array, String]] $dev_packages                             = $apache::params::dev_packages,
+  Optional[String] $ip                                                       = undef,
+  Boolean $service_enable                                                    = true,
+  Boolean $service_manage                                                    = true,
+  Variant[Boolean, Enum['running', 'stopped']] $service_ensure               = 'running',
```

Comment:
You could reuse a type
```suggestion
  Variant[Boolean, Stdlib::Ensure::Service] $service_ensure               = 'running',
```

I'd also consider dropping the `Boolean`.

Reply:
👍